README.md

```diff
@@ -1,2 +1,418 @@
+    {!shared.user.authorized && (
+      <button onClick={User.Logic.Login}>Login</button>
+    )}
+    <SearchForm.View model={model.form} />
+    <SearchResilts.View model={model.results} />
+  </div>
+)
+```
+**`const View = ({ model, shared })`** now also have a `shared` field in the props which is just a model of a shared block.
+
+### App depends on Shared (and how to disable it)
+It is important to notice that every view from app's block tree depends on changes of model of a shared tree. For example, when user loggs in (`authorazed` changes to `true` by `Login` command) each View of the app (of each block from app tree) will be re-rendered. For instance `Main.View`, `SearchForm.View`, `SearchResults.View`, all of them will be re-rendered.
+
+That imposes an important restriction to shared blocks. They should be changed rarely to have a good performance. But there is also a way to disable shared dependency for a specific block:
+
+```js
+// SearchResults.js
+...
+export const Logic = {
+  config({ subscribe, shared }) {
+    return {
+      manualSharedSubscribe: true,
+      subscriptions: [ subscribe(this.HandleUserUpdate(), shared.user) ]
+    }
+  },
+
+  @Cmd.batch
+  HandleUserUpdate(ctx, cmd, userModel) {
+  }
+  ...
+}
+```
+
+**`manualSharedSubscribe: true`** disables View re-rendering of `SearchResults` block when shared model changed.
+
+**`subscriptions: ...`** provides a way to subscribe only to some of shared model changes. Above we subscribed to changes of `shared.user` and added an update handler command `HandleUserUpdate`. This command will be executed on every change of `shared.user`. But not on changes of `shared` model itself.
+
+### Dealing with the world
+We showed above how to handle events from a View. But complex applications could have to handle not only view events, but some more, like WebSocket messages, or presses to `esc` in window scope, or execute someting in interval. The good news is that MJS also provides a way to handle this kind of events.
+
+```js
+// SearchResults.js
+...
+export const Logic = {
+  ...
+  port({ model, destroy, exec }) {
+    const timer = setInterval(() => {
+      exec(this.Search(model.query));
+    }, 10000);
+    destroy.then(() => clearInterval(timer));
+  }
+  ...
+}
+```
+**`port({ model, destroy, exec })`** is a special function of logic object, that is aimed to subscribe to some global events. In the example we just made an interval for refreshing the search results every 10 secs. Also we subscribed to the destroy Promise, which will be resolved when the block will be removed.
+
+## Conclusion
+That is all about basics of MJS. It was inspired by many existing frameworks/languages that the author used for a while. So probably there is no something extremly new. MJS is all about defining scalable, flexible way of implementing logic of your app in MVC manner with help of Command Pattern and of the latest available ES6/ES7 features, like decorators or async/await.
```

Comment:
"These are the basics of MJS" (you already repeat 'is all about' in the other sentence below)
...there is not anything extremely new.
' defining **a** scalable, flexible way...
'following the MVC pattern' or  'according to the MVC pattern'
'with **the** help of **the** Command Pattern"